ttnn/cpp/ttnn/operations/eltwise/binary_ng/device/binary_ng_program_factory.cpp

```diff
@@ -0,0 +1,401 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "binary_ng_device_operation.hpp"
+#include "tt_metal/common/work_split.hpp"
+#include "ttnn/operations/cb_utils.hpp"
+
+namespace {
+namespace CMAKE_UNIQUE_NAMESPACE {
+
+using namespace ttnn::operations::binary_ng;
+
+auto extract_shape_dims(const ttnn::SimpleShape& shape) {
+    return std::tuple{
+        shape.rank() >= 4 ? shape[-4] : 1u,      // N
+        shape.rank() >= 3 ? shape[-3] : 1u,      // C
+        shape[-2] / tt::constants::TILE_HEIGHT,  // Ht
+        shape[-1] / tt::constants::TILE_WIDTH    // Wt
```

Comment:
```suggestion
        shape[-3],      // N
        shape[-3],      // C
        shape[-2] / tt::constants::TILE_HEIGHT,  // Ht
        shape[-1] / tt::constants::TILE_WIDTH    // Wt
```

you can do that, since for safely we return 1 in case `-ID` is queried. But with positive index, it won't work.

Reply:
will do